executor/point_get_test.go

```diff
@@ -56,3 +56,242 @@ func (s *testSuite1) TestPointGet(c *C) {
 		`5 5 6 5 6 7 6 7 7`,
 	))
 }
+
+func (s *testSuite1) TestPointGetCharPK(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec(`use test;`)
+	tk.MustExec(`drop table if exists t;`)
+	tk.MustExec(`create table t(a char(2) primary key, b char(2));`)
+	tk.MustExec(`insert into t values("aa", "bb");`)
+
+	// Test truncate without sql mode `PAD_CHAR_TO_FULL_LENGTH`.
+	tk.MustExec(`set @@sql_mode="";`)
+	tk.MustPointGet(`select * from t where a = "aa";`).Check(testkit.Rows(`aa bb`))
+	tk.MustPointGet(`select * from t where a = "aab";`).Check(testkit.Rows())
+
+	// Test truncate with sql mode `PAD_CHAR_TO_FULL_LENGTH`.
+	tk.MustExec(`set @@sql_mode="PAD_CHAR_TO_FULL_LENGTH";`)
+	tk.MustPointGet(`select * from t where a = "aa";`).Check(testkit.Rows(`aa bb`))
+	tk.MustPointGet(`select * from t where a = "aab";`).Check(testkit.Rows())
+
+	tk.MustExec(`truncate table t;`)
+	tk.MustExec(`insert into t values("a ", "b ");`)
+
+	// Test trailing spaces without sql mode `PAD_CHAR_TO_FULL_LENGTH`.
+	tk.MustExec(`set @@sql_mode="";`)
+	tk.MustPointGet(`select * from t where a = "a";`).Check(testkit.Rows(`a b`))
+	tk.MustPointGet(`select * from t where a = "a ";`).Check(testkit.Rows())
+	tk.MustPointGet(`select * from t where a = "a  ";`).Check(testkit.Rows())
+
+	// Test trailing spaces with sql mode `PAD_CHAR_TO_FULL_LENGTH`.
+	tk.MustExec(`set @@sql_mode="PAD_CHAR_TO_FULL_LENGTH";`)
+	tk.MustPointGet(`select * from t where a = "a";`).Check(testkit.Rows())
```

Comment:
In my MySQL 5.7.25, this SQL returns `a b` 😱 ...
```
ERROR 1054 (42S22): Unknown column 'version' in 'field list'
mysql> select version();
+-----------+
| version() |
+-----------+
| 5.7.25    |
+-----------+
1 row in set (0.00 sec)

mysql> set @@sql_mode="PAD_CHAR_TO_FULL_LENGTH";
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t where a = "a";
+----+------+
| a  | b    |
+----+------+
| a  | b    |
+----+------+
1 row in set (0.00 sec)
```

Reply:
I think it's a bug according to the mysql 5.7 document, and it's fixed in mysql 8.0:
```sql
MySQL(root@localhost:test) > drop table t;
Query OK, 0 rows affected (0.11 sec)

MySQL(root@localhost:test) > create table t(a char(2) primary key, b char(2));
Query OK, 0 rows affected (0.06 sec)

MySQL(root@localhost:test) > insert into t values("a ", "b ");
Query OK, 1 row affected (0.03 sec)

MySQL(root@localhost:test) > set @@sql_mode="PAD_CHAR_TO_FULL_LENGTH";
Query OK, 0 rows affected (0.01 sec)

MySQL(root@localhost:test) > select * from t where a = "a";
Empty set (0.00 sec)

MySQL(root@localhost:test) > select version();
+-----------+
| version() |
+-----------+
| 8.0.11    |
+-----------+
1 row in set (0.00 sec)
```

Reply:
In my local MySQL 8.0.12, all tests below return `a, b`:
```
	tk.MustExec(`truncate table t;`)
	tk.MustExec(`insert into t values("a ", "b ");`)

	// Test trailing spaces without sql mode `PAD_CHAR_TO_FULL_LENGTH`.
	tk.MustExec(`set @@sql_mode="";`)
	tk.MustPointGet(`select * from t where a = "a";`).Check(testkit.Rows(`a b`))
	tk.MustPointGet(`select * from t where a = "a ";`).Check(testkit.Rows())
	tk.MustPointGet(`select * from t where a = "a  ";`).Check(testkit.Rows())

	// Test trailing spaces with sql mode `PAD_CHAR_TO_FULL_LENGTH`.
	tk.MustExec(`set @@sql_mode="PAD_CHAR_TO_FULL_LENGTH";`)
	tk.MustPointGet(`select * from t where a = "a";`).Check(testkit.Rows())
	tk.MustPointGet(`select * from t where a = "a ";`).Check(testkit.Rows(`a b`))
	tk.MustPointGet(`select * from t where a = "a  ";`).Check(testkit.Rows())
```

And in MySQL document: `As of MySQL 8.0.13, PAD_CHAR_TO_FULL_LENGTH is deprecated. It will be removed in a future version of MySQL.`

So it seems that MySQL 8.0.11~8.0.13 have different behavior on `PAD_CHAR_TO_FULL_LENGTH `.......... 🤣

Reply:
Orz